eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -361,9 +361,9 @@ object MultiPartHandler {
                     })
                   }
                 })).map(paths => {
-                  val invoiceFeatures = featuresTrampolineOpt.remove(Features.RouteBlinding).add(Features.RouteBlinding, FeatureSupport.Mandatory)
-                  val invoice = Bolt12Invoice(r.offer, r.invoiceRequest, paymentPreimage, r.nodeKey, nodeParams.channelConf.minFinalExpiryDelta, invoiceFeatures, paths.map { case (blindedRoute, paymentInfo, _) => PaymentBlindedRoute(blindedRoute.route, paymentInfo) })
-                  log.debug("generated invoice={} for offerId={}", invoice.toString, r.offer.offerId)
+                  val invoiceFeatures = nodeParams.features.bolt12Features().remove(Features.RouteBlinding).add(Features.RouteBlinding, FeatureSupport.Mandatory)
```

Comment:
If you apply https://github.com/ACINQ/eclair/commit/9bf9941b9b3bb3d073baef5d65bd19d063010c4e, you'll be able to remove that hack

```suggestion
                  val invoiceFeatures = nodeParams.features.bolt12Features()
```

Reply:
Done